packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/property-table/use-on-cell-edited.ts

```diff
@@ -0,0 +1,59 @@
+import { EditableGridCell, Item } from "@glideapps/glide-data-grid";
+import { useCallback } from "react";
+import { useBlockProtocolUpdateEntity } from "../../../../../components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolUpdateEntity";
+import { useSnackbar } from "../../../../../components/hooks/useSnackbar";
+import { useEntityEditor } from "../entity-editor-context";
+import { propertyGridIndexes } from "./constants";
+import { PropertyRow } from "./types";
+
+export const useOnCellEdited = (rowData: PropertyRow[]) => {
+  const snackbar = useSnackbar();
+  const { entity, setEntity } = useEntityEditor();
+  const { updateEntity } = useBlockProtocolUpdateEntity();
+
+  const onCellEdited = useCallback(
+    async ([col, row]: Item, newValue: EditableGridCell) => {
+      if (!entity) {
+        return;
+      }
+
+      const key = propertyGridIndexes[col];
+      const property = rowData[row];
+
+      if (!key || !property) {
+        throw new Error();
+      }
+
+      const updatedProperties = {
+        ...entity.properties,
+        [property.propertyTypeId]: newValue.data,
+      };
+
+      /**
+       * setting state for optimistic update
+       * also storing previous entity, so we can rollback if API call fails
+       */
+      const prevEntity = { ...entity };
```

Comment:
Why do you need to clone it?

Reply:
Hmm, you're right, calling setEntity on the line below does not change the value of `entity` immediately in this scope. I'll remove the clone, but I'll keep `prevEntity`, so it's more clear that we'Re rolling back to previous version if optimistic update fails

Reply:
Sure, but you can still put a string describing the error in the event it does happen.

Reply:
I guess you mention the `throw new Error()`s above

Reply:
Oh sorry this went to the wrong place – damn github app. Sorry!